### PR TITLE
[android] #2616 add MapActivity to SDK

### LINKED ISSUE #2616: MapActivity in SDK
Acknowledged by both @bleege and @ljbade that this would be a nice addition to the SDK.
Current setup requires overriding multiple lifecycle methods of Activity. 
In the past I forgot to add one or I had overridden the wrong one.. imo this setup is error prone. 
Besides that I would also simplify the code in the test application, which makes it easier to understand, 

Proposition:
 - in similar fashion as [ListActivitity] (https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/app/ListActivity.java) in the Android SDK
  - uses a default layout file, allows setting a different one but enforces a certain id (android.R.id.list)

### PR DESCRIPTION
**Pull request by tobrun**
Closes #2616 

Needs to be validated and documented before merging

### CODE REVIEW

**Comment by tobrun** [comment]
Have added some documentation, would love some :eyes:  on this @zugaldia @bleege

File: android/MapboxGLAndroidSDKTestApp/src/main/res/values/developer-config.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <string name="access_token">null</string>
+    <string name="access_token">pk.eyJ1IjoibnVyYm90IiwiYSI6Ijc5MjY3MGJlMGQzYzEwY2M1MjA3NDgzNjc4OTJjODNkIn0.eJOob0OKSWhuoPp_E3_ZDg</string>
```
Comment (bleege) [inline android/MapboxGLAndroidSDKTestApp/src/main/res/values/developer-config.xml]: @tobrun Need to keep the access_token out of Git.

Comment (tobrun) [inline android/MapboxGLAndroidSDKTestApp/src/main/res/values/developer-config.xml]: @bleege 
:facepalm:

**Comment by tobrun** [comment]
Closing this down hence issue is going to be closed

File: android/MapboxGLAndroidSDKTestApp/src/main/res/values/developer-config.xml (hunk 1 above)
Comment (bleege) [inline android/MapboxGLAndroidSDKTestApp/src/main/res/values/developer-config.xml]: :smile: These things happen.

**Pull request closed without merging**